app/models/manageiq/providers/azure/cloud_manager.rb

```diff
@@ -119,26 +117,32 @@ def vm_reboot_guest(vm, _options = {})
   end
 
   def vm_create_evm_snapshot(vm, options = {})
-    conf     = connect(options)
-    vm_svc   = vm.provider_service(conf)
-    snap_svc = snapshot_service(conf)
-    vm_obj   = vm_svc.get(vm.name, vm.resource_group)
-    return unless vm_obj.managed_disk?
-    os_disk      = vm_obj.properties.storage_profile.os_disk
+    @connection = connect(options)
+    vm.provider_service(@connection)
+    if vm.managed_disk?
+      vm_create_evm_managed_snapshot(vm)
+    else
+      vm_create_evm_blob_snapshot(vm)
+    end
+  end
+
+  def vm_create_evm_managed_snapshot(vm)
```

Comment:
Someday, we should move all of these `vm_*` methods into the their corresponding VM objects (via mixin or whatever is appropriate). This is a general comment that applies to all providers, and shouldn't be addressed in this PR. When this is addressed, we should probably also change the `evm` in names to something like `ssa`.

Reply:
Agreed.  There were suggestions made by @chessbyte on a previous PR in app/models/vm_scan.rb that should probably tackled at the same time.  We should add (an) issue(s) to track this.